kano_updater/commands/install.py

```diff
@@ -225,7 +225,10 @@ def install_pip_packages(progress):
 
         success = run_pip_command("install --upgrade {}".format(pkg))
 
-        # TODO: abort the install?
         if not success:
             msg = "Installing the '{}' pip package failed".format(pkg)
             logger.error(msg)
+            if not is_internet():
+                msg="Network is down, aborting PIP install"
+                logger.error(msg)
+                raise IOError(msg)
```

Comment:
@tombettany Maybe I'm not getting the context, but don't we basically have
to assume that the debian packages depend on the pip packages? It which
case it might not be a good idea to install them. I agree with the rest of
your comment though.

On 10 April 2015 at 10:43, tombettany <notifications@github.com> wrote:

> In kano_updater/commands/install.py
> <https://github.com/KanoComputing/kano-updater/pull/95#discussion_r28132781>
> :
>
> >          if not success:
> >              msg = "Installing the '{}' pip package failed".format(pkg)
> >              logger.error(msg)
> > +            if not is_internet():
> > +                msg="Network is down, aborting PIP install"
> > +                logger.error(msg)
> > +                raise IOError(msg)
>
> Hmm, do we want to end everything - if they don't have internet but have
> already downloaded the debian packages then these can be installed. Maybe
> we should skip.
>
> Also, in theory, because we cache the downloads in commands/download.py
> <https://github.com/KanoComputing/kano-updater/blob/master/kano_updater/commands/download.py#L133-L157>
> there should be a way of just installing the cached version without needing
> internet; I realise that the operation that we do here does require
> internet but the entire install process shouldn't and if we can find a way
> of not requiring it here then that would be ideal.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KanoComputing/kano-updater/pull/95/files#r28132781>.
>